be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/ElectionHandler.scala

```diff
@@ -41,25 +42,27 @@ class ElectionHandler(dbRef: => AskableActorRef) extends MessageHandler {
   /** Overrides default DbActor with provided parameter
     */
   override final val dbActor: AskableActorRef = dbRef
+  private val serverUnexpectedAnswer: String = "The server is doing something unexpected"
 
   def handleSetupElection(rpcMessage: JsonRpcRequest): GraphMessage = {
     // FIXME: add election info to election channel/electionData
-    val message: Message = rpcMessage.getParamsMessage.get
-    val data: SetupElection = message.decodedData.get.asInstanceOf[SetupElection]
-    val electionId: Hash = data.id
-    val electionChannel: Channel = Channel(s"${rpcMessage.getParamsChannel.channel}${Channel.CHANNEL_SEPARATOR}$electionId")
-    val keyPair = KeyPair()
 
     // need to write and propagate the election message
     val combined = for {
+      (_, message, somedata) <- extractParameters[SetupElection](rpcMessage, serverUnexpectedAnswer)
+      data = somedata.get
+      electionId: Hash = data.id
+      laoId: Hash = data.loa
+      electionChannel: Channel = Channel(s"${rpcMessage.getParamsChannel.channel}${Channel.CHANNEL_SEPARATOR}$electionId")
+      keyPair = KeyPair()
       _ <- dbActor ? DbActor.WriteAndPropagate(rpcMessage.getParamsChannel, message)
       _ <- dbActor ? DbActor.CreateChannel(electionChannel, ObjectType.ELECTION)
       _ <- dbActor ? DbActor.WriteAndPropagate(electionChannel, message)
-      _ <- dbActor ? DbActor.CreateElectionData(electionId, keyPair)
-    } yield ()
+      _ <- dbActor ? DbActor.CreateElectionData(electionId, keyPair, laoId)
```

Comment:
Since when does `ElectionData` take `laoId` as a parameter? Be careful to not overlap with other issues you are working on.